Expensify/App
# Make sure report appears for deep links that are not the Home route

### Details
### Fixed Issues
Regression from code removed in https://github.com/Expensify/Expensify.cash/pull/2346/files

### Tests
### QA Steps
1. Navigate directly to `/settings`
2. Verify that the report  + home screen is loaded in the background

### Tested On

- [ ] Desktop // No deep linking on Desktop so there's no need to test

### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
